### DISCUSSION TITLE
Awake action for Cinnamon

### DISCUSSION THREAD

**Discussion by ivanednor98** [foresto/joystickwake#16] [Q&A] [answered]
Hi all, I tried to use joystickwake on Cinnamon installed on top of Ubuntu and after 1 hour with the gamepad configured in my settings, the display turned off. Could you please suggest a command that could keep the display on, thanks in advance.

**Comment by foresto** [comment, marked as answer]
Can you try running it like this? [Edit: This doesn't work; please install python3-dbus-next as described below.]

    joystickwake --loglevel info --command 'cinnamon-screensaver-command --deactivate'

If that works, you can put `command = cinnamon-screensaver-command --deactivate` in $HOME/.config/joystickwake/joystickwake.conf to apply it to all future joystickwake runs.

And, if you verify that it works, I'll add it to the list of default wakers so no config will be needed in future versions.

**Reply by foresto** [reply]
...And if it doesn't work, please copy the output here so I can have a look at what joystickwake has to say on your system.

**Reply by foresto** [reply]
On a Linux Mint 21.3 VM, `cinnamon-screensaver-command --deactivate` does wake the screen, and its options appear to be modeled after gnome-screensaver-command's traditional behavior. However, it doesn't behave consistently with that behavior, or consistently at all:

In my tests, it left the screensaver timer unaffected in some cases (leading the screen to blank again shortly after the wake) and apparently left the timer disabled in other cases (leaving the screen awake forever). Since this behavior is not useful, I don't think I'll add that command to joystickwake's defaults.

Inspecting its freedesktop ScreenSaver d-bus interface, I see that the SimulateUserActivity method is present, but actually calling it yields a "not supported" error. So that's useless as well.

That leaves one option: Joystickwake's recently added dbus org.freedesktop.ScreenSaver.Inhibit support. This requires the python3-dbus-next package to be installed on your system. It looks like it appeared upstream in Ubuntu 23.10 (Mantic), which is newer than the current Mint release, but based on the timeline, I *think* it should be available in the upcoming Linux Mint release (Wilma), and in the current Linux Mint Debian Edition (Faye) release.

If you get that package installed, joystickwake should automatically detect it and enable the dbus Inhibit functionality.

**Reply by foresto** [reply]
Update: Linux Mint 22 (Wilma) was released today, and has the python3-dbus-next package. Please let me know if it helps.

**Comment by ivanednor98** [comment]
Hi @foresto, thanks a lot for the reply, I tried this command `joystickwake --command 'cinnamon-screensaver-command --deactivate'` and it did prevent the screen from shutting down due to inactivity, but when I closed the game I found 2 identical Ubuntu internal error reports:

![image](https://github.com/user-attachments/assets/2e02cc2e-84b6-4aa4-95a0-4e1d7ed8efb8)

**Reply by foresto** [reply]
Looks like a bug in cinnamon-screensaver or cinnamon-screensaver-command. Between that, the useless behavior I described earlier, and the fake d-bus interfaces, it seems Cinnamon has some work to do on their screen saver. If you're inclined, you might report those errors to their maintainers.

As for joystickwake, it looks like your remaining option is to install the python3-dbus-next package, as described above.
